docs/migration_guide.rst

```diff
@@ -1,10 +1,55 @@
+X.X.X migration guide
+=======================
+
+TODO(Update this with correct version or move to separate file)
+
+This version deprecates the AccountClient, which is a major change to the StarkNet.py
+It is being replaced with new :class:`BaseAccount <starknet_py.net.account.base_account.BaseAccount>` ABC and its
+default implementation :class:`Account <starknet_py.net.account.account.Account>`.
+
+Unlike ``AccountClient``, an ``Account`` is not a Client anymore. That means that methods like
+``get_storage_at``, ``call_contract`` etc. are not available in the Account interface.
+
+However ``Account`` now exposes a `.`client`` property, which means using an ``Account`` is
+just as simple as ``AccountClient`` was. For example:
+
+.. code-block:: python
+
+    account_client.get_storage_at(contract_address=0x1234)
+    # becomes
+    account.client.get_storage_at(contract_address=0x1234)
+
+Replacing inheritance with composition simplifies the ``Account`` interface and will make
+maintaining ``Account`` simpler.
+
+Changes in the Account interface
+--------------------------------
+
+1. Removed ``hash_message`` method. Use ``TypedData.message_hash`` directly instead.
+2. ``Account`` doesn't expose a ``net`` property
+3. ``Account`` doesn't accept a ``supported_tx_version`` parameter. It currently always uses version 1.
```

Comment:
Any ideas how we want to handle new tx versions?

Reply:
I though that if we have a account interface anyways, we can have separate implementations (or subclasses of Account) for different versions if the need arises. That way we wouldn't have to deal with spaghetti version handling like we do in AccountClient, and could deprecated older versions easily.

Reply:
Ok, alias will work